monero/exports.sh

```diff
@@ -0,0 +1,111 @@
+			"stagenet")
+				MONERO_NETWORK="stagenet";;
+		esac
+	fi
+} > /dev/null || true
+
+
+
+if [[ ! -f "${MONERO_ENV_FILE}" ]]; then
+	if [[ -z "${MONERO_NETWORK}" ]]; then
+		MONERO_NETWORK="mainnet"
+	fi
+	
+	if [[ -z ${MONERO_RPC_USER+x} ]] || [[ -z ${MONERO_RPC_PASS+x} ]] || [[ -z ${MONERO_RPC_AUTH+x} ]]; then
+		MONERO_RPC_USER="monero"
+		MONERO_RPC_DETAILS=$("${EXPORTS_APP_DIR}/scripts/rpcauth.py" "${MONERO_RPC_USER}")
+		MONERO_RPC_PASS=$(echo "$MONERO_RPC_DETAILS" | tail -1)
+		MONERO_RPC_AUTH=$(echo "$MONERO_RPC_DETAILS" | head -2 | tail -1 | sed -e "s/^rpc-login=//")
+	fi
+
+	echo "export APP_MONERO_NETWORK='${MONERO_NETWORK}'"		>  "${MONERO_ENV_FILE}"
+	echo "export APP_MONERO_RPC_USER='${MONERO_RPC_USER}'"	>> "${MONERO_ENV_FILE}"
+	echo "export APP_MONERO_RPC_PASS='${MONERO_RPC_PASS}'"	>> "${MONERO_ENV_FILE}"
+	echo "export APP_MONERO_RPC_AUTH='${MONERO_RPC_AUTH}'"	>> "${MONERO_ENV_FILE}"
+fi
+
+. "${MONERO_ENV_FILE}"
+
+# Make sure we don't persist the original value in .env if we have a more recent
+# value from the app config
+{
+	if [[ ! -z ${MONERO_NETWORK+x} ]] && [[ "${MONERO_NETWORK}" ]] && [[ "${APP_MONERO_NETWORK}" ]]
+	then
+		APP_MONERO_NETWORK="${MONERO_NETWORK}"
+	fi
+} > /dev/null || true
+
+if [[ "${APP_MONERO_NETWORK}" == "mainnet" ]]; then
+	MONERO_CHAIN="mainnet"
+elif [[ "${APP_MONERO_NETWORK}" == "testnet" ]]; then
+	MONERO_CHAIN="testnet"
+	export APP_MONERO_RPC_PORT="28081"
+	export APP_MONERO_P2P_PORT="28080"
+
+elif [[ "${APP_MONERO_NETWORK}" == "stagenet" ]]; then
+	MONERO_CHAIN="stagenet"
+	export APP_MONERO_RPC_PORT="38081"
+	export APP_MONERO_P2P_PORT="38080"
+else
+	echo "Warning (${EXPORTS_APP_ID}): Monero Network '${APP_MONERO_NETWORK}' is not supported"
+fi
+
+export MONERO_DEFAULT_NETWORK="${MONERO_CHAIN}"
+
+BIN_ARGS=()
+# Commenting out options that are replaced by generated config file. We should migrate all these over in a future update.
+BIN_ARGS+=( "--rpc-bind-port=${APP_MONERO_RPC_PORT}" )
+BIN_ARGS+=( "--rpc-bind-ip=0.0.0.0" )
+BIN_ARGS+=( "--confirm-external-bind" )
+
+BIN_ARGS+=( "--rpc-login=\"${APP_MONERO_RPC_AUTH}\"" )
+
+export APP_MONERO_COMMAND=$(IFS=" "; echo "${BIN_ARGS[@]}")
+
+# echo "${APP_MONERO_COMMAND}"
+
+rpc_hidden_service_file="${EXPORTS_TOR_DATA_DIR}/app-${EXPORTS_APP_ID}-rpc/hostname"
+p2p_hidden_service_file="${EXPORTS_TOR_DATA_DIR}/app-${EXPORTS_APP_ID}-p2p/hostname"
+export APP_MONERO_RPC_HIDDEN_SERVICE="$(cat "${rpc_hidden_service_file}" 2>/dev/null || echo "notyetset.onion")"
+export APP_MONERO_P2P_HIDDEN_SERVICE="$(cat "${p2p_hidden_service_file}" 2>/dev/null || echo "notyetset.onion")"
+
+{
+	# Migrate settings for app updates differently to fresh installs
+	MONERO_INSTALL_EXISTS="false"
+	MONERO_DATA_DIR="${EXPORTS_APP_DIR}/data/monero"
+	if [[ -d "${MONERO_DATA_DIR}/lmdb" ]]
+	then
+	MONERO_INSTALL_EXISTS="true"
+	fi
+
+	APP_CONFIG_EXISTS="false"
+	if [[ -f "${EXPORTS_APP_DIR}/data/app/monero-config.json" ]]
+	then
+	APP_CONFIG_EXISTS="true"
+	fi
+} || true
```

Comment:
Are you using this block? This was created for the Bitcoin Node app when we introduced "Advanced Settings", where we didn't want to clobber existing user's manual settings. Since it looks like Advanced Settings has always been part of your Monero App, you can likely remove this and associated code in the forked umbrel-bitcoin repo.

Reply:
Removing and resolving as dead code. Thanks!